### PR TITLE
[GPU] Fix ConvolutionKernel_b_fs_yx_fsv16_1x1 to support input0 feature dynamic case

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Fix ConvolutionKernel_b_fs_yx_fsv16_1x1 to support input0 feature dynamic case

### Tickets:
 - 146681

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp [hunk 1]
```diff
@@ -307,8 +307,13 @@ JitConstants ConvolutionKernel_b_fs_yx_fsv16_1x1::GetJitConstants(const convolut
         jit.AddConstant(MakeJitConstant("OUTPUT_LEFTOVERS", output_leftover));
 
         const auto input_leftover_num = "(" + input0_dims.f() + "%" + feature_block_size + ")";
-        const auto input_leftover = "(" + input_leftover_num + "!= 0)";
-        jit.AddConstant(MakeJitConstant("INPUT_LEFTOVERS", input_leftover));
+        if (!params.inputs[0].Feature().is_dynamic) {
```
Comment (hyunback) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp:310]: Frankly, I don't understand why added to check whether dynamic or static?
As I know, we can check whether dynamic or not in Line 237 and SA kernel can run leftover or not depends on feature size from input param in runtime.

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp]: Updated to fix in cl kernel code side not here. Thanks!

**Review by hyunback — APPROVED** [review]

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/convolution_gpu_bfyx_f16_1x1.cl [hunk 2]
```diff
@@ -122,6 +122,28 @@ KERNEL(convolution_b_fs_yx_fsv16_1x1)(
...
+                    const uint yb = y + (x + i) / INPUT0_SIZE_X;
+                    const uint input_idx = input_offset + k * input_fs_pitch + yb * input_y_pitch + xb * input_x_pitch;
+
+                    src[i] = input[input_idx + sglid];
+                }
+#else
+                src = input[input_offset + k * input_fs_pitch + sglid];
+#endif // X_BLOCK_SIZE > 1
+            }
+        }
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/convolution_gpu_bfyx_f16_1x1.cl:144]: this code block seems identical to line 148~165. I guess we can adjust the condition to simplify the code..

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/convolution_gpu_bfyx_f16_1x1.cl]: Updated it. Thanks!

**Review by isanghao — APPROVED** [review]

**Pull request merged**